Src/Core/Collections/CollectionAssertions.cs

```diff
@@ -1096,21 +1063,26 @@ public AndConstraint<TAssertions> StartWith(object element, string because = "",
         /// </param>
         public AndConstraint<TAssertions> EndWith(object element, string because = "", params object[] becauseArgs)
         {
+            AssertCollectionEndsWith(Subject?.Cast<object>(), new[] { element }, ObjectExtensions.IsSameOrEqualTo, because, becauseArgs);
+            return new AndConstraint<TAssertions>((TAssertions) this);
+        }
+
+        protected void AssertCollectionEndsWith<TActual, TExpected>(IEnumerable<TActual> actual, TExpected[] expected, Func<TActual, TExpected, bool> predicate, string because = "", params object[] reasonArgs)
+        {
             Execute.Assertion
-                .BecauseOf(because, becauseArgs)
-                .WithExpectation("Expected {context:collection} to end with {0}{reason}, ", element)
-                .ForCondition(!ReferenceEquals(Subject, null))
-                .FailWith("but the collection is {0}.", (object)null)
+                .BecauseOf(because, reasonArgs)
+                .WithExpectation("Expected {context:collection} to end with {0}{reason}, ", expected)
+                .Given(() => actual)
+                .AssertCollectionIsNotNullOrEmpty()
```

Comment:
I like the way you extend the fluent API here although I'm not entirely sure the naming is correct since the API is there to build up an assertion, e.g. *execute an assertion because of some reason with expectation*. Unfortunately, I can't come up with anything better.